typescript/cli/src/commands/warp.ts

```diff
@@ -427,6 +448,236 @@ export const check: CommandModuleWithContext<{
   },
 };
 
+export const rebalancer: CommandModuleWithWriteContext<{
+  config: string;
+  checkFrequency: number;
+  withMetrics: boolean;
+  monitorOnly: boolean;
+  manual?: boolean;
+  origin?: string;
+  destination?: string;
+  amount?: string;
+}> = {
+  command: 'rebalancer',
+  describe: 'Run a warp route collateral rebalancer',
+  builder: {
+    config: {
+      type: 'string',
+      description:
+        'The path to a rebalancer configuration file (.json or .yaml)',
+      demandOption: true,
+      alias: ['rebalancerConfigFile', 'rebalancerConfig', 'configFile'],
+    },
+    checkFrequency: {
+      type: 'number',
+      description: 'Frequency to check balances in ms (defaults: 30 seconds)',
+      demandOption: false,
+      default: 60000,
+    },
+    withMetrics: {
+      type: 'boolean',
+      description: 'Enable metrics (default: true)',
+      demandOption: false,
+      default: false,
+    },
+    monitorOnly: {
+      type: 'boolean',
+      description: 'Run in monitor only mode (default: false)',
+      demandOption: false,
+      default: false,
+    },
+    manual: {
+      type: 'boolean',
+      description:
+        'Trigger a rebalancer manual run (default: false, requires --origin, --destination, --amount)',
+      demandOption: false,
+      implies: ['origin', 'destination', 'amount'],
+    },
+    origin: {
+      type: 'string',
+      description: 'The origin chain for manual rebalance',
+      demandOption: false,
+      implies: 'manual',
+    },
+    destination: {
+      type: 'string',
+      description: 'The destination chain for manual rebalance',
+      demandOption: false,
+      implies: 'manual',
+    },
+    amount: {
+      type: 'number',
+      description:
+        'The amount to transfer from origin to destination on manual rebalance. Defined in token units (E.g 100 instead of 100000000 wei for USDC)',
+      demandOption: false,
+      implies: 'manual',
+    },
+  },
+  handler: async ({
```

Comment:
By analogy with the `warp deploy` command, it would be great if all the logic under `handle()` could move to somewhere under `typescript/cli/src/rebalancer`. This way, a bigger part of our rebalancing code will live under one directory.